platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,381 @@
+
+To ensure that one build won't override files in the other, we'll have to
+1. Update
+   [`vets-website-deploy.sh`](https://github.com/department-of-veterans-affairs/devops/blob/2c63e6d62fa5ad2b6b87478e30d0a7bc2820c252/ansible/deployment/roles/deploy-vets-website/files/vets-website-deploy.sh)
+   to copy only to the `/applications/` directory
+1. Create a new deploy script for the content deployment which copies everything
+   over _except_ the `/applications/` directory
+
+#### Testing changes to applications
+- The Webpack configuration will use
+  [`html-webpack-plugin`](https://github.com/jantimon/html-webpack-plugin) to
+  generate temporary landing pages for React applications
+  - This will happen for all build types
+    - Engineers will be able to access the site on a `localhost` build
+    - CI can run E2E tests against production code without the output of the
+      content build
+- The E2E test script in CI will start a process to serve these pages so the
+  tests can run a browser and access the applications
+- **Important note:** These HTML files will not be served in production; the
+  landing pages for the applications come from the content build
+  - To ensure these pages aren't made in production, the deploy script will copy
+    only the files in `/applications/`
+
+#### Miscellaneous
+- We need a separate content validation job that runs once a day on a schedule
+  - **Question:** How will we validate links to `vets-website` assets?
+  - **Question:** Where will this job be run? Jenkins? Nomad? Circle?
+- The application build will be responsible for creating the `settings.js`
+  - It's currently a step in the Metalsmith script
+- The landing pages to the applications will be created in the CMS build
+- The CMS repo will no longer import `vets-website`, but rather the new content
+  build repo
+  
+#### React application landing pages
+**Important note:** JavaScript application landing pages will be created during
+the content build. There will be a JSON file with the parameters to create the
+landing pages. Parameters include:
+- URL of the page
+- Bundle name for the application
+- Whether the file should be made available in production
+
+When we do this, we need to update the manifest.json documentation and remove
+unused properties from all the files.
+
+**Question:** Should this be a transitional stopgap measure until the landing
+pages are all created in Drupal, or could this be a viable long-term solution?
+
+#### Cache busting application code
+To ensure the latest code is downloaded by the browser, we have to bust the
+browser's cache. **Currently**, we're adding hashes to the asset file names and
+changing the `<script>` tag's `src` property to point to the file name with
+hash.
+
+Once we split out the content build, however, it's infeasible to continue this
+approach. Instead, we'll rely on the (`Cache-Control`
+header)[https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control].
```

Comment:
Anything to add to this, @rianfowler?

Reply:
We need to summarize the trade off of using the `cache-control` header:

browsers will always make a request before using their cache for the main document
  - higher traffic for S3
  - lower performance for users
- new deployments will be available to browsers only after the the cache expiration time

Reply:
> new deployments will be available to browsers only after the the cache expiration time

Is that...true? With `no-cache`, the browser will always make the request (and get back a `304 Not Modified` if it hasn't been modified since cached, I believe), so browsers should always get the latest deployments. Unless I'm misunderstanding something, anyhow.

I'll add the other trade-offs, though. :+1: